lib/AST/NameLookup.cpp

```diff
@@ -2812,28 +2872,53 @@ createOpaqueParameterGenericParams(
     if (!typeRepr)
       continue;
 
-    auto opaqueTypeReprs = typeRepr->collectOpaqueReturnTypeReprs();
-    for (auto opaqueRepr : opaqueTypeReprs) {
+    // Plain protocols should imply 'some' with experimetal feature
+    CollectedOpaqueReprs typeReprs;
+    if (ctx.LangOpts.hasFeature(Feature::ImplicitSome)) {
+        typeReprs = collectTypeReprs(typeRepr);
+    } else {  typeReprs = collectOpaqueReturnTypeReprs(typeRepr); }
```

Comment:
Should we check for `Feature::ImplicitSome` inside of `collectOpaqueReturnTypeReprs` instead of having two separate functions here? I think that should be pretty straightforward, since we only want to collect `OpaqueReturnTypeRepr` if the experimental feature is not enabled.